pkg/operator/oidcdiscoveryendpoint/controller.go

```
@@ -0,0 +1,424 @@
+				r.logger.WithField("bucket", bucketName).Info("bucket already exists and is owned by us")
+			default:
+				return aerr
+			}
+		} else {
+			return err
+		}
+	} else {
+		r.logger.WithField("bucket", bucketName).Info("bucket created")
+	}
+
+	// Tag bucket for deprovisioning
+	_, err = awsClient.PutBucketTagging(&s3.PutBucketTaggingInput{
+		Bucket: awssdk.String(bucketName),
+		Tagging: &s3.Tagging{
+			TagSet: []*s3.Tag{
+				{
+					Key:   awssdk.String("kubernetes.io/cluster/" + r.infrastructureName),
+					Value: awssdk.String("owned"),
+				},
+			},
+		},
+	})
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			return aerr
+		}
+		return err
+	}
+	r.logger.WithField("bucket", bucketName).Debug("bucket tagged")
+
+	// Render and create the OIDC discovery JSON s3 object in the well-known location
+	issuerURL := r.getIssuerURL()
+	discoveryJSON := fmt.Sprintf(discoveryTemplate, issuerURL, issuerURL, keysURI)
+	_, err = awsClient.PutObject(&s3.PutObjectInput{
+		ACL:    awssdk.String("public-read"),
+		Body:   awssdk.ReadSeekCloser(strings.NewReader(discoveryJSON)),
+		Bucket: awssdk.String(bucketName),
+		Key:    awssdk.String(discoveryURI),
+	})
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			return aerr
+		}
+		return err
+	}
+	r.logger.WithField("bucket", bucketName).Debug("discovery document updated")
+
+	// Extract the token signer and create keys JSON s3 object for jwks_uri
+	tokenSignerConfigMap, err := r.kubeclientset.CoreV1().ConfigMaps(tokenSignerConfigMapNamespace).Get(context.TODO(), tokenSignerConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	pemKeys := [][]byte{}
+	for _, pemKey := range tokenSignerConfigMap.Data {
+		pemKeys = append(pemKeys, []byte(pemKey))
+	}
+	if len(pemKeys) < 1 {
+		return fmt.Errorf("no signing keys found in config map %s/%s", tokenSignerConfigMapNamespace, tokenSignerConfigMapName)
+	}
+
+	keysJSON, err := encodeKeys(pemKeys)
```

error is not handled.